debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/DumpModuleCommand.java

```diff
@@ -0,0 +1,240 @@
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.PackageIteratorFilter;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.PackageOutput;
+
+import static com.ibm.j9ddr.vm29.tools.ddrinteractive.commands.DumpModuleCommand.Status.*;
+
+public class DumpModuleCommand extends Command 
+{
+	enum Status {
+		CLASS, MODULE, PACKAGE, ALL, HELP
+	}
+
+	public DumpModuleCommand()
+	{
+		addCommand("dumpmodule", "[all|requires|exports|classes|packages] <moduleAddress>|help", "List details about a provided module");
+	}
+	
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException 
+	{
+		Status status = HELP;
+		ClassIteratorFilter classFilter = null;
+		ClassOutput classOutput = null;
+		ModuleIteratorFilter moduleFilter = null;
+		ModuleOutput moduleOutputter = null;
+		PackageIteratorFilter packageFilter = null;
+		PackageOutput packageOutputter = null;
+		String filterArg = null;
+		switch (args.length) {
+		case 1:
+			if (!args[0].equalsIgnoreCase("help")) {
+				filterArg = args[0];
+				status = ALL;
+			} /* else stay status.HELP */
+			break;
+		case 2:
+			filterArg = args[1];
+			switch (args[0]) {
+			case "all":
+				status = ALL;
+				break;
+			case "requires":
+				moduleFilter = DumpModuleCommand::filterModuleRequired;
+				moduleOutputter = ModularityHelper::printJ9Module;
+				status = MODULE;
+				break;
+			case "exports":
+				packageFilter = DumpModuleCommand::filterPackageModuleAndExport;
+				packageOutputter = ModularityHelper::printPackageExportVerbose;
+				status = PACKAGE;
+				break;
+			case "classes":
+				classFilter = DumpModuleCommand::filterClassByModule;
+				classOutput = ModularityHelper::printJ9Class;
+				status = CLASS;
+				break;
+			case "packages":
+				packageFilter = DumpModuleCommand::filterPackageModule;
+				packageOutputter = ModularityHelper::printJ9Package;
+				status = PACKAGE;
+				break;
+			default:
+				break; /* default stay status.HELP */
+			}
+			break;
+		default:
+			break; /* default stay status.HELP */
+		}
+		try {
+			int result;
+			J9ModulePointer modulePtr = J9ModulePointer.cast(Long.decode(filterArg));
+			J9ClassLoaderPointer classLoaderPtr = modulePtr.classLoader();;
+			switch (status) {
+			case CLASS:
+				result = ModularityHelper.iterateClassLoaderClasses(out, classFilter, classOutput, filterArg, classLoaderPtr);
+				System.out.printf("Found %d class%s\n", result, (1 == result ? "" : "es"));
+				break;
+			case MODULE:
+				result = ModularityHelper.iterateModules(out, moduleFilter, moduleOutputter, filterArg);
+				System.out.printf("Found %d module%s\n", result, (1 == result ? "" : "s"));
+				break;
+			case PACKAGE:
+				result = ModularityHelper.iterateClassLoaderPackages(out, packageFilter, packageOutputter, filterArg, classLoaderPtr);
+				System.out.printf("Found %d package%s\n", result, (1 == result ? "" : "s"));
+				break;
+			case ALL:
+				out.printf("Module:\n");
```

Comment:
Don't use `printf` where `println` would suffice; and don't use `\n` as it's not necessarily the right line separator.
Where `printf` is appropriate, use `%n` to obtain a newline.